crates/oxc_linter/src/rules/jsx_a11y/label_has_associated_control.rs

```diff
@@ -0,0 +1,1560 @@
+            config.depth = std::cmp::min(depth, 25).try_into().unwrap();
+        }
+
+        if let Some(assert) = options.get("assert").and_then(serde_json::Value::as_str) {
+            config.assert = match assert {
+                "htmlFor" => Assert::HtmlFor,
+                "nesting" => Assert::Nesting,
+                "both" => Assert::Both,
+                _ => Assert::Either,
+            };
+        }
+
+        if let Some(label_components) =
+            options.get("labelComponents").and_then(serde_json::Value::as_array)
+        {
+            if let Some(mut components) = label_components
+                .iter()
+                .map(serde_json::Value::as_str)
+                .map(|component| component.map(std::string::ToString::to_string))
+                .collect::<Option<Vec<String>>>()
+            {
+                config.label_components.append(&mut components);
+            }
+        }
+
+        if let Some(label_attributes) =
+            options.get("labelAttributes").and_then(serde_json::Value::as_array)
+        {
+            if let Some(mut attributes) = label_attributes
+                .iter()
+                .map(serde_json::Value::as_str)
+                .map(|attribute| attribute.map(std::string::ToString::to_string))
+                .collect::<Option<Vec<String>>>()
+            {
+                config.label_attributes.append(&mut attributes);
+            }
+        }
+
+        if let Some(control_components) =
+            options.get("controlComponents").and_then(serde_json::Value::as_array)
+        {
+            control_components.iter().map(serde_json::Value::as_str).for_each(|component| {
+                let Some(component) = component else {
+                    return;
+                };
+
+                let Ok(glob) = Glob::new(component) else {
+                    return;
+                };
+
+                control_builder.add(glob);
+            });
+        }
+
+        config.control_components = if let Ok(controls) = control_builder.build() {
+            controls
+        } else {
+            let mut control_builder = GlobSetBuilder::new();
+            control_builder.add(Glob::new("input").unwrap());
+            control_builder.add(Glob::new("meter").unwrap());
+            control_builder.add(Glob::new("output").unwrap());
+            control_builder.add(Glob::new("progress").unwrap());
+            control_builder.add(Glob::new("select").unwrap());
+            control_builder.add(Glob::new("textarea").unwrap());
+            control_builder.build().unwrap()
+        };
+
+        Self(Box::new(config))
+    }
+
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::JSXElement(element) = node.kind() else {
+            return;
+        };
+
+        if let Some(element_type) = get_element_type(ctx, &element.opening_element) {
+            if !self.0.label_components.contains(&element_type.to_string()) {
```

Comment:
Consider implementing `Deref` for your rule, so you don't have to do `self.0` all the time.

Reply:
Thanks for the tip! Done